### PR TITLE
[Draft] Add configure process with custom patch

### PR DESCRIPTION
**Pull request by Hhhilulu**
# Description
- Configure process with custom patch.

Usage:
Add `patch_dict `and `configure_custom_patch` in your test case.
The `patch_dict`'s `key `is your mocked methods path, and the `value` is your mocked methods.
image
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by thy09 — DISMISSED** [review]

File: src/promptflow/tests/executor/conftest.py [hunk 1]
```diff
@@ -40,27 +41,19 @@ def recording_setup():
             patcher.stop()
 
 
-def _default_mock_process_wrapper(*args, **kwargs):
-    # Default mock implementation of _process_wrapper in recording mode
-    setup_recording()
-    _process_wrapper(*args, **kwargs)
+def _custom_mock_process_wrapper(*args, **kwargs):
```
Comment (chw-microsoft) [inline src/promptflow/tests/executor/conftest.py:44]: 1. add more comments to this function
2. consider use explicit args

File: src/promptflow/tests/executor/conftest.py [hunk 2]
```diff
@@ -74,6 +67,34 @@ def process_override():
                     multiprocessing.Process = original_process_class[start_method]
 
 
+def setup_custom_process_target(patch_dict=None):
```
Comment (chw-microsoft) [inline src/promptflow/tests/executor/conftest.py:70]: setup_process_targets()

File: src/promptflow/tests/executor/conftest.py [hunk 3]
```diff
@@ -87,6 +108,11 @@ def recording_injection(recording_setup, process_override):
         recording_array_reset()
 
 
+@pytest.fixture
+def configure_custom_patch(patch_dict):
+    yield from configure_process_with_custom_patch(patch_dict)
```
Comment (chw-microsoft) [inline src/promptflow/tests/executor/conftest.py:113]: rename --> configure_process_with_custom_patch

File: src/promptflow/tests/executor/conftest.py
Comment (chw-microsoft) [inline src/promptflow/tests/executor/conftest.py:1]: Please move public utitlies into process_utils.py with more comments

**Pull request closed without merging**